spec/components/filter_section_spec.rb

```diff
@@ -0,0 +1,41 @@
+require "spec_helper"
+
+describe "Filter section component", type: :view do
+  def component_name
+    "filter_section"
+  end
+
+  def render_component(locals)
+    render "components/#{component_name}", locals
+  end
+
+  it "raises an error if title option is omitted" do
+    expect { render_component({}) }.to raise_error(/title/)
+  end
+
+  it "renders a filter section that is closed" do
+    render_component({ title: "title" })
+
+    assert_select ".app-c-filter-section", count: 1
+    assert_select ".app-c-filter-section[open=open]", count: 0
+  end
+
+  it "renders a filter section that is open when open option is true" do
+    render_component({ title: "title", open: true })
+
+    assert_select ".app-c-filter-section[open=open]"
+  end
+
+  it "set section title text with hidden context for accessibility" do
+    render_component({ title: "section title" })
+
+    assert_select ".app-c-filter-section__summary-title", text: "filter by section title"
+    assert_select ".app-c-filter-section__summary-title .govuk-visually-hidden", text: "filter by"
+  end
+
+  it "set section status text" do
```

Comment:
Is it worth testing that the status text message doesn't get rendered if it isn't given?